STYLE.md

```diff
@@ -0,0 +1,531 @@
+`unsigned short`.
+
+Sizes should be represented as `size_t`. When converting to/from an
+`int` for legacy purposes ensure to account for overflow/underflow
+conditions.
+
+Within structs that are retained across the lifetime of a connection,
+for new integer values whose size are known and it's easy to do, use a
+smaller integer type like `uint8_t`. This is a "free" connection
+footprint optimization for servers. Don't make code significantly more
+complex for it, and do still check the bounds when passing in and out
+of the struct. This narrowing should not propagate to local variables
+and function parameters which should use more conventional integer
+types in order to not add complexity for the users of such
+functions. Do not retroactively apply this rule to existing integer
+values in structures as this could cause ABI breakage.
+
+When doing arithmetic, account for overflow conditions.
+
+Except in platform specific code, do not use `ssize_t`. MSVC lacks it,
+and prefer out-of-band error signaling for `size_t` (see Return
+Values).
+
+## Functions
+
+Ideally, functions should be short and sweet, and do just
+one thing. A rule of thumb is that they should fit on one or two
+screenfuls of text (25 lines as we all know), and do one thing and do
+that well.
+
+The maximum length of a function is often inversely proportional to
+the complexity and indentation level of that function. So, if you have
+a conceptually simple function that is just one long (but simple)
+switch statement, where you have to do lots of small things for a lot
+of different cases, it’s okay to have a longer function.
+
+If you have a complex function, however, consider using helper
+functions with descriptive names. You can ask the compiler to in-line
+them if you think it’s performance-critical, and it will probably do a
+better job of it than you would have done.
+
+Another measure of complexity is the number of local variables. If
+there are more than five to 10, consider splitting it into smaller
+pieces. A human brain can generally easily keep track of about seven
+different things; anything more and it gets confused. Often things
+which are simple and clear now are much less obvious two weeks from
+now, or to someone else. An exception to this is the command-line
+applications which support many options.
+
+In source files, separate functions with one blank line.  In function
+prototypes, include parameter names with their data types. Although
+this is not required by the C language, it is preferred in OpenSSL
+because it is a simple way to add valuable information for the
+reader. The name in the prototype declaration should match the name in
+the function definition.
+
+### Checking function arguments
+
+A public function should verify that its arguments are sensible. This
+includes, but is not limited to, verifying that:
+
+    * non-optional pointer arguments are not NULL and
+    * numeric arguments are within expected ranges.
```

Comment:
Markdown understand indentation of 4 or more spaces as a code block.  I assume that's not what you wanted here.